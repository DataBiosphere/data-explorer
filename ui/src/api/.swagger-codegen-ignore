docs/**
git_push.sh
mocha.opts
.travis.yml
