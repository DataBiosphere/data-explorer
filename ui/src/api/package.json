{
  "name": "data_explorer_service",
  "version": "0.0.1",
  "description": "API_Service_that_reads_from_Elasticsearch_",
  "license": "Unlicense",
  "main": "src/index.js",
  "scripts": {
    "test": "mocha --compilers js:babel-core/register --recursive"
  },
  "browser": {
    "fs": false
  },
  "dependencies": {
    "babel": "^6.23.0",
    "babel-cli": "^6.24.1",
    "superagent": "3.5.2"
  },
  "devDependencies": {
    "babel-core": "6.18.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "expect.js": "~0.3.1",
    "mocha": "~2.3.4",
    "sinon": "1.17.3"
  }
}
