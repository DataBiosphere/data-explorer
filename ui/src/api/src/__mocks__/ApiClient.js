import { getFacetsList } from "../../../../tests/mockutils";
import FacetsResponse from "../model/FacetsResponse";

export const mockCallApi = jest.fn(
  (
    path,
    httpMethod,
    pathParams,
    queryParams,
    headerParams,
    formParams,
    bodyParam,
    authNames,
    contentTypes,
    accepts,
    returnType,
    callback
  ) => {
    if (path === "facets" && httpMethod === "GET") {
      let getFacetsResponse = new FacetsResponse();
      getFacetsResponse.count = 210;
      getFacetsResponse.facets = getFacetsList(5, 6);
      return getFacetsResponse;
    }
    return null;
  }
);
const mock = jest.fn().mockImplementation(() => {
  return {
    callApi: mockCallApi,
    buildCollectionParam: jest.fn()
  };
});

export default mock;
