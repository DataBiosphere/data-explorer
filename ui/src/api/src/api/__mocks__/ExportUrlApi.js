import ExportUrlResponse from "../../model/ExportUrlResponse";

export const mockExportUrlPost = jest.fn(callback => {
  let postExportUrlResponse = new ExportUrlResponse();
  postExportUrlResponse.url = "exportUrl";
  callback(null, postExportUrlResponse);
  return postExportUrlResponse;
});

const mock = jest.fn().mockImplementation(() => {
  return {
    exportUrlPost: mockExportUrlPost
  };
});

export default mock;
