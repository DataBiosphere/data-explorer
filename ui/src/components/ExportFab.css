.exportFab {
  right: 20px;
  bottom: 20px;
  position: fixed;
}
