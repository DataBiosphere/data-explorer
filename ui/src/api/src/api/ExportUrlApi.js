/**
 * Data Explorer Service
 * API Service that reads from Elasticsearch.
 *
 * OpenAPI spec version: 0.0.1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import ExportUrlRequest from '../model/ExportUrlRequest';
import ExportUrlResponse from '../model/ExportUrlResponse';

/**
* ExportUrl service.
* @module api/ExportUrlApi
* @version 0.0.1
*/
export default class ExportUrlApi {

    /**
    * Constructs a new ExportUrlApi. 
    * @alias module:api/ExportUrlApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the exportUrlPost operation.
     * @callback module:api/ExportUrlApi~exportUrlPostCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ExportUrlResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Creates and returns a signed URL to a GCS zip file of JSON files. The JSON files represent entities to be exported to a Terra workspace. https://app.terra.bio/#import-data may be called with the url parameter set to this url. For each JSON file, https://rawls.dsde-prod.broadinstitute.org/#!/entities/create_entity may be called with the JSON as the POST body. 
     * @param {Object} opts Optional parameters
     * @param {module:model/ExportUrlRequest} opts.exportUrlRequest 
     * @param {module:api/ExportUrlApi~exportUrlPostCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ExportUrlResponse}
     */
    exportUrlPost(opts, callback) {
      opts = opts || {};
      let postBody = opts['exportUrlRequest'];


      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = [];
      let contentTypes = [];
      let accepts = [];
      let returnType = ExportUrlResponse;

      return this.apiClient.callApi(
        '/exportUrl', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }


}
