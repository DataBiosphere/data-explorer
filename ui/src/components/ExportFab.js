/** Export to Terra FAB */

import CloudUpload from "@material-ui/icons/CloudUpload";
import Button from "@material-ui/core/Button";
import Tooltip from "@material-ui/core/Tooltip";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import React from "react";

import "components/ExportFab.css";

class ExportFab extends React.Component {
  constructor(props) {
    super(props);
    this.state = { open: false };
    this.handleClick = this.handleClick.bind(this);
    this.handleSave = this.handleSave.bind(this);
    this.handleCancel = this.handleCancel.bind(this);
    this.setTextValue = this.setTextValue.bind(this);
  }

  render() {
    return (
      <div>
        {/*
          Style div instead of button itself, to prevent button from moving
          when cohort dialog is shown. See
          https://github.com/mui-org/material-ui/issues/9275#issuecomment-350479467
        */}
        <div className="mui-fixed exportFab">
          <Tooltip title="Send to Terra">
            <Button
              variant="fab"
              color="secondary"
              onClick={() => this.handleClick()}
            >
              <CloudUpload />
            </Button>
          </Tooltip>
        </div>
        <div>
          <Dialog
            open={this.state.open}
            onClose={this.handleClose}
            aria-labelledby="form-dialog-title"
          >
            <DialogContent>
              <TextField
                autoFocus
                onChange={this.setTextValue}
                margin="dense"
                id="name"
                label="Cohort Name"
                helperText="A cohort with this name will be created in Terra"
                type="text"
                fullWidth
                onKeyPress={ev => {
                  if (ev.key === "Enter") {
                    this.handleSave();
                  }
                }}
              />
            </DialogContent>
            <DialogActions>
              <Button onClick={this.handleCancel} color="primary">
                Cancel
              </Button>
              <Button id="save" onClick={this.handleSave} color="primary">
                Send
              </Button>
            </DialogActions>
          </Dialog>
        </div>
      </div>
    );
  }

  setTextValue(event) {
    this.setState({ cohortName: event.target.value });
  }

  handleClick() {
    var filter = this.props.filter;
    if (filter != null && filter.length > 0) {
      this.setState(state => ({ open: true }));
    } else {
      this.handleSave();
    }
  }

  handleCancel() {
    this.setState(state => ({ open: false }));
  }

  handleSave() {
    this.setState(state => ({ open: false }));
    let exportUrlCallback = function(error, data) {
      if (error) {
        alert(error.response.body.detail);
      } else {
        let importUrl =
          "https://app.terra.bio/#import-data?format=entitiesJson";
        if (data.authorization_domain) {
          importUrl += "&ad=" + data.authorization_domain;
        }
        importUrl += "&url=" + data.url;
        window.location.assign(importUrl);
      }
    }.bind(this);
    let cohortName = this.state.cohortName;
    let filter = this.props.filter;
    if (filter == null) {
      filter = [];
    }
    if (cohortName == null) {
      cohortName = "";
    }
    let params = new Object();
    params.cohortName = cohortName;
    params.filter = filter;
    this.props.exportUrlApi.exportUrlPost(
      { exportUrlRequest: params },
      exportUrlCallback
    );
  }
}

export default ExportFab;
