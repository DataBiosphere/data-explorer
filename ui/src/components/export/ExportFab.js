/** Export to Saturn FAB */

import "./ExportFab.css";

import React from "react";
import CloudUpload from "@material-ui/icons/CloudUpload";
import Button from "@material-ui/core/Button";
import ExportForm from "./ExportForm";

class ExportFab extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dialogOpen: false
    };
    this.handleClickOpen = this.handleClickOpen.bind(this);
    this.handleClose = this.handleClose.bind(this);
  }

  handleClickOpen() {
    this.setState({ dialogOpen: true });
  }

  handleClose() {
    this.setState({ dialogOpen: false });
  }

  render() {
    return (
      <div>
        <Button
          variant="fab"
          color="secondary"
          className="exportFab"
          onClick={this.handleClickOpen}
        >
          <CloudUpload />
        </Button>
        <ExportForm
          open={this.state.dialogOpen}
          handleClose={this.handleClose}
        />
      </div>
    );
  }
}

export default ExportFab;
