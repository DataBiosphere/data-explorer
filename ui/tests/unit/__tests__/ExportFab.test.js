import React from "react";
import ExportFab from "../../../src/components/ExportFab";
import { getMuiTheme } from "material-ui/styles/index";
import { mockExportUrlPost } from "../../../src/api/src/api/ExportUrlApi";
import Button from "@material-ui/core/Button";
import PropTypes from "prop-types";
import ExportUrlApi from "../../../src/api/src/api/ExportUrlApi";
import ApiClient from "../../../src/api/src/ApiClient";

jest.mock("../../../src/api/src/api/ExportUrlApi");
("use strict");

beforeEach(() => {
  mockExportUrlPost.mockClear();
});

test("Renders correctly", () => {
  const tree = shallow(
    <ExportFab exportUrlApi={new ExportUrlApi(new ApiClient())} />
  );
  expect(tree).toMatchSnapshot();
});
