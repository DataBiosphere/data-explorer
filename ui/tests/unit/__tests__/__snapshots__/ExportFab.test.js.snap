// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<WithStyles(Tooltip)
  title="Export to Saturn"
>
  <WithStyles(Button)
    className="exportFab"
    color="secondary"
    variant="fab"
  >
    <pure(CloudUpload) />
  </WithStyles(Button)>
</WithStyles(Tooltip)>
`;
